Consul/Client_GetRequests.cs

```diff
@@ -171,6 +172,27 @@ protected override void ApplyOptions(ConsulClientConfiguration clientConfig)
             {
                 Params["near"] = Options.Near;
             }
+
+            if (Options.UseCache && Options.Consistency != ConsistencyMode.Consistent)
```

Comment:
I think it is better not to check the consistency mode. It is up to a consul server to validate arguments and fail if the combination of arguments is invalid. 
```suggestion
            if (Options.UseCach)
```

Reply:
This is what the original Go API does but it definitely makes more sense to let it fail than possibly stay silent after using incorrect `QueryOptions`